__tests__/server/middleware/csp.spec.js

```diff
@@ -67,6 +72,39 @@ describe('csp', () => {
       expect(headers).not.toHaveProperty('content-security-policy-report-only');
     });
 
+    it('adds ip and localhost to csp in development', () => {
+      process.env.NODE_ENV = 'development';
+      process.env.HTTP_ONE_APP_DEV_CDN_PORT = 5000;
+      const requiredCsp = requireCSP();
+      const cspMiddleware = requiredCsp.default;
+      const { updateCSP } = requiredCsp;
+      updateCSP("default-src 'none'; script-src 'self'; connect-src 'self';");
+      cspMiddleware()(req, res, next);
+      // eslint-disable-next-line no-underscore-dangle
+      const headers = res._getHeaders();
+      expect(headers).toHaveProperty('content-security-policy');
+      const cspString = headers['content-security-policy'];
+      expect(sanitizeCspString(cspString)).toMatchSnapshot();
+    });
+
+    it('does not add ip and localhost to csp in production', () => {
+      process.env.NODE_ENV = 'production';
+      delete process.env.HTTP_ONE_APP_DEV_CDN_PORT;
+      const requiredCsp = requireCSP();
+      const cspMiddleware = requiredCsp.default;
+      const { updateCSP } = requiredCsp;
+      updateCSP("default-src 'none'; script-src 'self'; connect-src 'self';");
+      cspMiddleware()(req, res, next);
+      // eslint-disable-next-line no-underscore-dangle
+      const headers = res._getHeaders();
+      expect(headers).toHaveProperty('content-security-policy');
+      const cspString = headers['content-security-policy'];
+      const ipFound = cspString.match(/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}:3001/);
```

Comment:
maybe just test for ip, and ignore the port

Reply:
https://github.com/americanexpress/one-app/pull/42#discussion_r384167132

Reply:
fixed in 0cee717